### ISSUE TITLE
Incompatibility with mod Levels [no grave key on respawn]

### ISSUE THREAD

**Issue by D3nnis3n** [Corail31/tombstone_lite#2]
Hello, the mod seems to work great - but i do not get a tombstone key after dying. Also, despite the setting for decaytime being -1 i can open the tomb about five minutes later without a key. (So i know it works in base)

Sometimes when i die i receive a key while still being in the menu screen where i can press respawn. This is not the case everytime, but sometimes. As soon as i press "Respawn" the key has gone away.

I use 1.10.2 and we play on a SMP Server.

**Comment by Corail31** [comment]
is it the beta or 1.2.0?

**Comment by D3nnis3n** [comment]
It is 1.2.1, 1.2.0 did not work at all for me, it crashes on my server.

**Comment by Corail31** [comment]
version of forge?

**Comment by D3nnis3n** [comment]
1.10.2-12.18.3.2281

**Comment by D3nnis3n** [comment]
I might also be misunderstanding something, i should receive a tomb key on death, right? If yes, this is not the case.

**Comment by Corail31** [comment]
you should receive a key is the option needAccess is set on server side !

**Comment by Corail31** [comment]
so if the server is configure to need no key and a decay time of 5min , it'll be this for all the players in

**Comment by D3nnis3n** [comment]
This is my servers configuration file:
```
# Configuration file

~CONFIG_VERSION: 1.2.1

general {
    # Time before tomb doesn't need key [range: -1 ~ 50000, default: -1]
    I:decayTime=-1

    # Highlight the tomb from far when holding the key [default: true]
    B:highlight=true

    # Ingredient to upgrade the key [default: minecraft:ender_pearl:1]
    S:ingredientToUpgradeKey=minecraft:ender_pearl:1

    # Percent of xp lost [range: 0 ~ 100, default: 0]
    I:percentXpLoss=0

    # Show fog around the graves [default: true]
    B:showFog=true

    # Allow teleport to dimension with the key [default: true]
    B:teleportDim=true

    # Need a key to open the tomb [default: true]
    B:tombAccess=true

    # Tomb's key upgradable [default: true]
    B:upgradeTombKey=true

    # Experience loss [default: false]
    B:xpLoss=false
}
```

**Comment by Corail31** [comment]
I'll make a test on server 1.10.2 in a few minutes, but for me this is strange.

**Comment by D3nnis3n** [comment]
Well, it already works better than version 1.2.0 - that just outright crashed the server (but not the singleplayer!). I wanted to create a issue for that, too, but then the new version already was out and that worked so far.

**Comment by Corail31** [comment]
for me the mod has no bug... and 1.2.0 doesn't crash or i wouldn't have publish it...
so i have to understand how you do this on your server

**Comment by Corail31** [comment]
is there bug log on server?

**Comment by D3nnis3n** [comment]
Sure, please note i use Sponge and MoCreatures on the live server. I also tried it without both - this does not change anything. This is the error i receive after death, but i do not know if this has anything to do with it:

```
> [13:22:34] [Server thread/INFO]: D3nnis3n was slain by lion
> [13:22:34] [Server thread/ERROR] [FML]: Exception caught during firing event net                                                                                                                                                                                                                                                                          .minecraftforge.event.entity.player.PlayerDropsEvent@ca34812:
> java.lang.NullPointerException: Unknown statistic: achievement.FirstTomb
>         at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:                                                                                                                                                                                                                                                                          253) ~[minecraft_server.1.10.2.jar:?]
>         at net.minecraft.util.text.event.HoverEvent.getHandle(SourceFile:572) ~[                                                                                                                                                                                                                                                                          fa.class:?]
>         at net.minecraft.util.text.TextComponentBase.toText(TextComponentBase.ja                                                                                                                                                                                                                                                                          va:710) ~[ew.class:?]
>         at net.minecraft.util.text.TextComponentTranslation.wrapFormatArgs(Sourc                                                                                                                                                                                                                                                                          eFile:570) ~[ff.class:?]
>         at net.minecraft.util.text.TextComponentTranslation.createBuilder(Source                                                                                                                                                                                                                                                                          File:552) ~[ff.class:?]
>         at net.minecraft.util.text.TextComponentBase.toText(TextComponentBase.ja                                                                                                                                                                                                                                                                          va:697) ~[ew.class:?]
>         at org.spongepowered.common.text.SpongeTexts.toText(SpongeTexts.java:79)                                                                                                                                                                                                                                                                           ~[SpongeTexts.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.text.chat.ChatUtil.sendMessage(ChatUtil.java                                                                                                                                                                                                                                                                          :48) ~[ChatUtil.class:1.10.2-2281-5.2.0-BETA-2274]
>         at net.minecraft.server.management.PlayerList.func_148544_a(PlayerList.j                                                                                                                                                                                                                                                                          ava:2074) ~[mp.class:?]
>         at net.minecraft.server.management.PlayerList.func_148539_a(PlayerList.j                                                                                                                                                                                                                                                                          ava:1070) ~[mp.class:?]
>         at net.minecraft.stats.StatisticsManagerServer.func_150873_a(SourceFile:                                                                                                                                                                                                                                                                          69) ~[nr.class:?]
>         at net.minecraft.stats.StatisticsManager.func_150871_b(SourceFile:38) ~[                                                                                                                                                                                                                                                                          nx.class:?]
>         at net.minecraft.entity.player.EntityPlayerMP.func_71064_a(EntityPlayerM                                                                                                                                                                                                                                                                          P.java:945) ~[lu.class:?]
>         at ovh.corail.tombstone.handler.EventHandler.onLivingDrops(EventHandler.                                                                                                                                                                                                                                                                          java:65) ~[EventHandler.class:?]
>         at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_283_EventH                                                                                                                                                                                                                                                                          andler_onLivingDrops_LivingDropsEvent.invoke(.dynamic) ~[?:?]
>         at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASM                                                                                                                                                                                                                                                                          EventHandler.java:90) ~[ASMEventHandler.class:?]
>         at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.jav                                                                                                                                                                                                                                                                          a:635) ~[EventBus.class:?]
>         at org.spongepowered.mod.event.SpongeForgeEventFactory.callLivingDropsEv                                                                                                                                                                                                                                                                          ent(SpongeForgeEventFactory.java:894) ~[SpongeForgeEventFactory.class:1.10.2-228                                                                                                                                                                                                                                                                          1-5.2.0-BETA-2274]
>         at org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(Sp                                                                                                                                                                                                                                                                          ongeForgeEventFactory.java:464) ~[SpongeForgeEventFactory.class:1.10.2-2281-5.2.                                                                                                                                                                                                                                                                          0-BETA-2274]
>         at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEvent                                                                                                                                                                                                                                                                          Manager.java:292) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEvent                                                                                                                                                                                                                                                                          Manager.java:338) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEvent                                                                                                                                                                                                                                                                          Manager.java:326) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:143) ~[                                                                                                                                                                                                                                                                          SpongeImpl.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.event.tracking.phase.entity.DeathPhase.lambd                                                                                                                                                                                                                                                                          a$unwind$4(DeathPhase.java:159) ~[DeathPhase.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.event.tracking.CapturedMultiMapSupplier.ifPr                                                                                                                                                                                                                                                                          esentAndNotEmpty(CapturedMultiMapSupplier.java:61) [CapturedMultiMapSupplier.cla                                                                                                                                                                                                                                                                          ss:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.event.tracking.phase.entity.DeathPhase.unwin                                                                                                                                                                                                                                                                          d(DeathPhase.java:143) [DeathPhase.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.event.tracking.phase.entity.EntityPhase.unwi                                                                                                                                                                                                                                                                          nd(EntityPhase.java:52) [EntityPhase.class:1.10.2-2281-5.2.0-BETA-2274]
>         at org.spongepowered.common.event.tracking.CauseTracker.completePhase(Ca                                                                                                                                                                                                                                                                          useTracker.java:223) [CauseTracker.class:1.10.2-2281-5.2.0-BETA-2274]
>         at net.minecraft.entity.EntityLivingBase.func_70097_a(EntityLivingBase.j                                                                                                                                                                                                                                                                          ava:3339) [sf.class:?]
>         at net.minecraft.entity.player.EntityPlayer.func_70097_a(EntityPlayer.ja                                                                                                                                                                                                                                                                          va:1022) [zs.class:?]
>         at net.minecraft.entity.player.EntityPlayerMP.func_70097_a(EntityPlayerM                                                                                                                                                                                                                                                                          P.java:543) [lu.class:?]
>         at drzhark.mocreatures.entity.MoCEntityAnimal.func_70652_k(MoCEntityAnim                                                                                                                                                                                                                                                                          al.java:1124) [MoCEntityAnimal.class:?]
>         at net.minecraft.entity.ai.EntityAIAttackMelee.func_190102_a(EntityAIAtt                                                                                                                                                                                                                                                                          ackMelee.java:150) [ty.class:?]
>         at net.minecraft.entity.ai.EntityAIAttackMelee.func_75246_d(EntityAIAtta                                                                                                                                                                                                                                                                          ckMelee.java:139) [ty.class:?]
>         at net.minecraft.entity.ai.EntityAITasks.func_75774_a(SourceFile:129) [t                                                                                                                                                                                                                                                                          r.class:?]
>         at net.minecraft.entity.EntityLiving.func_70626_be(EntityLiving.java:773                                                                                                                                                                                                                                                                          ) [sg.class:?]
>         at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.j                                                                                                                                                                                                                                                                          ava:2183) [sf.class:?]
>         at net.minecraft.entity.EntityLiving.func_70636_d(EntityLiving.java:577)                                                                                                                                                                                                                                                                           [sg.class:?]
>         at net.minecraft.entity.EntityAgeable.func_70636_d(EntityAgeable.java:15                                                                                                                                                                                                                                                                          5) [rt.class:?]
>         at net.minecraft.entity.passive.EntityAnimal.func_70636_d(SourceFile:41)                                                                                                                                                                                                                                                                           [wc.class:?]
>         at drzhark.mocreatures.entity.MoCEntityAnimal.func_70636_d(MoCEntityAnim                                                                                                                                                                                                                                                                          al.java:356) [MoCEntityAnimal.class:?]
>         at drzhark.mocreatures.entity.MoCEntityTameableAnimal.func_70636_d(MoCEn                                                                                                                                                                                                                                                                          tityTameableAnimal.java:415) [MoCEntityTameableAnimal.class:?]
>         at drzhark.mocreatures.entity.passive.MoCEntityBigCat.func_70636_d(MoCEn                                                                                                                                                                                                                                                                          tityBigCat.java:296) [MoCEntityBigCat.class:?]
>         at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.                                                                                                                                                                                                                                                                          java:2013) [sf.class:?]
>         at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295                                                                                                                                                                                                                                                                          ) [sg.class:?]
>         at org.spongepowered.common.event.tracking.TrackingUtil.tickEntity(Track                                                                                                                                                                                                                                                                          ingUtil.java:159) [TrackingUtil.class:1.10.2-2281-5.2.0-BETA-2274]
>         at net.minecraft.world.WorldServer.redirect$onCallEntityUpdate$zjp000(Wo                                                                                                                                                                                                                                                                          rldServer.java:2815) [ls.class:?]
>         at net.minecraft.world.World.func_72866_a(World.java:4072) [aid.class:?]
>         at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:839) [l                                                                                                                                                                                                                                                                          s.class:?]
>         at net.minecraft.world.World.func_72870_g(World.java:1934) [aid.class:?]
>         at net.minecraft.world.World.func_72939_s(World.java:5957) [aid.class:?]
>         at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:2157) [                                                                                                                                                                                                                                                                          ls.class:?]
>         at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.jav                                                                                                                                                                                                                                                                          a:709) [MinecraftServer.class:?]
>         at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(Dedicated                                                                                                                                                                                                                                                                          Server.java:387) [ld.class:?]
>         at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.jav                                                                                                                                                                                                                                                                          a:613) [MinecraftServer.class:?]
>         at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [M                                                                                                                                                                                                                                                                          inecraftServer.class:?]
>         at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
```

**Comment by Corail31** [comment]
You're right, i have to make some adjustments for 1.10.2 version, there's a trouble in the achievement handler :s The 1.11.2 doesn't have this trouble

**Comment by Corail31** [comment]
thanks i 'll work on it today :) i have to make the 1.2.2 release for 1.10.2 (but 1.10.2 make more errors than 1.11.2 :p)

**Comment by D3nnis3n** [comment]
You're welcome. 

Yeh, problem is all main mods are finally on 1.10.2 but only some of them are 1.11.2 - so i can't switch. Without Mekanism, EnderIO, ProjectRed, BaseMetals and so on, and so on.

But yes, the last 1.10.2 Version did not even start on servers - so i hope you will be able to fix that easily 🗡

**Comment by Corail31** [comment]
Always easy to fix, the difficulty is to make tests on each version and on server side, that makes lots of tests :) But i made tests :
ok : 1.2.0 crash server on 1.10.2
but : 1.2.1 i just made a test with the key and so on... all is working....

**Comment by Corail31** [comment]
@D3nnis3n can you confirm me this :
- The only times that you don't get the key was because you had anything to loose (any item) (or you were in creative mod or with the rule keep inventory set to true)
- The real bugs are 1.2.0 don"t work at all on server, and 1.2.1 has just achievements broken

**Comment by D3nnis3n** [comment]
> The only times that you don't get the key was because you had anything to loose (any item) (or you were in creative mod or with the rule keep inventory set to true)

I had a completley full inventory, was NOT in creative mod and the rule keep inventory is set to FALSE. So no, i can not confirm.

> The real bugs are 1.2.0 don"t work at all on server, and 1.2.1 has just achievements broken
I can not confirm. I can confirm 1.2.0 crashes on server and does not work at all and 1.2.1 has achievements broken (but i didn't notice that, as i don't care for achievements nor knew there were some).

**Comment by Corail31** [comment]
When i try 1.2.1, and i have only the message on server start about achievements, but then, all is working... so i can't reproduce the bug.

**Comment by Corail31** [comment]
can i send you a release to test?

**Comment by D3nnis3n** [comment]
Sure, you can.

**Comment by Corail31** [comment]
[https://github.com/Corail31/trash/blob/master/tombstone-1.2.2-1.10.2.jar?raw=true](url)

**Comment by D3nnis3n** [comment]
404 for that link.

**Comment by Corail31** [comment]
if you have a discord we can discuss there too :)

**Comment by D3nnis3n** [comment]
Unfortunatley not. I'll just test it as soon as the link works. :D

**Comment by Corail31** [comment]
retry the link :)

**Comment by D3nnis3n** [comment]
Ah, kay, pasting worked.

**Comment by Corail31** [comment]
if it don't work, i'll make a version that skip achievements for server purpose (for now)

**Comment by D3nnis3n** [comment]
My minecraft is starting up, will see it soon.

**Comment by Corail31** [comment]
if you want to see some features of the last release, just set decorative graves on the ground, and use command /weather thunder ;)

**Comment by D3nnis3n** [comment]
Unfortunatley it still does not work. Achievements still get an error in the log:
```
[19:40:57] [Server thread/INFO]: D3nnis3n was slain by Zombie
[19:40:57] [Server thread/ERROR] [FML]: Exception caught during firing event net.minecraftforge.event.entity.player.PlayerDropsEvent@66f4d43b:
java.lang.NullPointerException: Unknown statistic: achievement.firstTomb
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:253) ~[minecraft_server.1.10.2.jar:?]
        at net.minecraft.util.text.event.HoverEvent.getHandle(SourceFile:572) ~[fa.class:?]
        at net.minecraft.util.text.TextComponentBase.toText(TextComponentBase.java:710) ~[ew.class:?]
        at net.minecraft.util.text.TextComponentTranslation.wrapFormatArgs(SourceFile:570) ~[ff.class:?]
        at net.minecraft.util.text.TextComponentTranslation.createBuilder(SourceFile:552) ~[ff.class:?]
        at net.minecraft.util.text.TextComponentBase.toText(TextComponentBase.java:697) ~[ew.class:?]
        at org.spongepowered.common.text.SpongeTexts.toText(SpongeTexts.java:79) ~[SpongeTexts.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.text.chat.ChatUtil.sendMessage(ChatUtil.java:48) ~[ChatUtil.class:1.10.2-2281-5.2.0-BETA-2274]
        at net.minecraft.server.management.PlayerList.func_148544_a(PlayerList.java:2074) ~[mp.class:?]
        at net.minecraft.server.management.PlayerList.func_148539_a(PlayerList.java:1070) ~[mp.class:?]
        at net.minecraft.stats.StatisticsManagerServer.func_150873_a(SourceFile:69) ~[nr.class:?]
        at net.minecraft.stats.StatisticsManager.func_150871_b(SourceFile:38) ~[nx.class:?]
        at net.minecraft.entity.player.EntityPlayerMP.func_71064_a(EntityPlayerMP.java:945) ~[lu.class:?]
        at ovh.corail.tombstone.handler.EventHandler.onLivingDrops(EventHandler.java:65) ~[EventHandler.class:?]
        at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_283_EventHandler_onLivingDrops_LivingDropsEvent.invoke(.dynamic) ~[?:?]
        at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90) ~[ASMEventHandler.class:?]
        at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:635) ~[EventBus.class:?]
        at org.spongepowered.mod.event.SpongeForgeEventFactory.callLivingDropsEvent(SpongeForgeEventFactory.java:894) ~[SpongeForgeEventFactory.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.mod.event.SpongeForgeEventFactory.callForgeEvent(SpongeForgeEventFactory.java:464) ~[SpongeForgeEventFactory.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:292) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:338) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.mod.event.SpongeModEventManager.post(SpongeModEventManager.java:326) ~[SpongeModEventManager.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.SpongeImpl.postEvent(SpongeImpl.java:143) ~[SpongeImpl.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.event.tracking.phase.entity.DeathPhase.lambda$unwind$4(DeathPhase.java:159) ~[DeathPhase.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.event.tracking.CapturedMultiMapSupplier.ifPresentAndNotEmpty(CapturedMultiMapSupplier.java:61) [CapturedMultiMapSupplier.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.event.tracking.phase.entity.DeathPhase.unwind(DeathPhase.java:143) [DeathPhase.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.event.tracking.phase.entity.EntityPhase.unwind(EntityPhase.java:52) [EntityPhase.class:1.10.2-2281-5.2.0-BETA-2274]
        at org.spongepowered.common.event.tracking.CauseTracker.completePhase(CauseTracker.java:223) [CauseTracker.class:1.10.2-2281-5.2.0-BETA-2274]
        at net.minecraft.entity.EntityLivingBase.func_70097_a(EntityLivingBase.java:3339) [sf.class:?]
        at net.minecraft.entity.player.EntityPlayer.func_70097_a(EntityPlayer.java:1022) [zs.class:?]
        at net.minecraft.entity.player.EntityPlayerMP.func_70097_a(EntityPlayerMP.java:543) [lu.class:?]
        at net.minecraft.entity.monster.EntityMob.func_70652_k(SourceFile:602) [yx.class:?]
        at net.minecraft.entity.monster.EntityZombie.func_70652_k(EntityZombie.java:361) [zi.class:?]
        at net.minecraft.entity.ai.EntityAIAttackMelee.func_190102_a(EntityAIAttackMelee.java:150) [ty.class:?]
        at net.minecraft.entity.ai.EntityAIAttackMelee.func_75246_d(EntityAIAttackMelee.java:139) [ty.class:?]
        at net.minecraft.entity.ai.EntityAIZombieAttack.func_75246_d(SourceFile:28) [uy.class:?]
        at net.minecraft.entity.ai.EntityAITasks.func_75774_a(SourceFile:129) [tr.class:?]
        at net.minecraft.entity.EntityLiving.func_70626_be(EntityLiving.java:773) [sg.class:?]
        at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2183) [sf.class:?]
        at net.minecraft.entity.EntityLiving.func_70636_d(EntityLiving.java:577) [sg.class:?]
        at net.minecraft.entity.monster.EntityMob.func_70636_d(SourceFile:47) [yx.class:?]
        at net.minecraft.entity.monster.EntityZombie.func_70636_d(EntityZombie.java:279) [zi.class:?]
        at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2013) [sf.class:?]
        at net.minecraft.entity.EntityLiving.func_70071_h_(EntityLiving.java:295) [sg.class:?]
        at net.minecraft.entity.monster.EntityMob.func_70071_h_(SourceFile:52) [yx.class:?]
        at net.minecraft.entity.monster.EntityZombie.func_70071_h_(EntityZombie.java:356) [zi.class:?]
        at org.spongepowered.common.event.tracking.TrackingUtil.tickEntity(TrackingUtil.java:159) [TrackingUtil.class:1.10.2-2281-5.2.0-BETA-2274]
        at net.minecraft.world.WorldServer.redirect$onCallEntityUpdate$zjp000(WorldServer.java:2815) [ls.class:?]
        at net.minecraft.world.World.func_72866_a(World.java:4072) [aid.class:?]
        at net.minecraft.world.WorldServer.func_72866_a(WorldServer.java:839) [ls.class:?]
        at net.minecraft.world.World.func_72870_g(World.java:1934) [aid.class:?]
        at net.minecraft.world.World.func_72939_s(World.java:5957) [aid.class:?]
        at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:2157) [ls.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:709) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:387) [ld.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:613) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:471) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]

```

Also, i can now definatley confirm the key is there in respawn menu but has gone after i pressed "respawn". Maybe this is a failure of sponge? Could you try it with spongeforge?

**Comment by Corail31** [comment]
ah yeah i didn't think to sponge :s
can you send me the report on pastebin.com and remove this one plz ? :)

**Comment by Corail31** [comment]
i contact sponge staff for this :)

**Comment by D3nnis3n** [comment]
I found a sponge error afterwards, but those errors appear at nearly every mod - none didn't work because of this. Maybe you could ask there, i'm not very familiar with all of this.

https://pastebin.com/puhtQLe5

**Comment by D3nnis3n** [comment]
But would it not be a solution to give the key after respawn instead of after death? (Or something similar :D)

**Comment by D3nnis3n** [comment]
Otherwise: Sorry to have bugged you for a sponge error or something :O

**Comment by Corail31** [comment]
That's not a real bug except i don't handle the achievements on server :)
I'll make the achievements to be removed when on server and post release in one hour :)

**Comment by D3nnis3n** [comment]
I meant the thing with the key, that it is removed after pressing respawn while it is there in the respawn screen.

**Comment by Corail31** [comment]
the key need to know the position ;) so it's created there ;)

**Comment by D3nnis3n** [comment]
I see, i hope sponge can sort this out. Seems to remove everything after respawn. :/

**Comment by D3nnis3n** [comment]
Could you might consider an option that tombstone is not created when slain by player (for PVP loot?) - that would be awesome.

**Comment by Corail31** [comment]
Don't worry, i'll patch this :) with or without achievement :p
For looting, there's the "no need access option"

**Comment by D3nnis3n** [comment]
Nah, i only want players to be able to loot the grave when slain by other player - but not when the player has died by mobs, environment, etc. We got a PVPvE Server. Items shall be safe when in PvE, but not in PvP.

**Comment by Corail31** [comment]
ah yeah, could be possible so if it's about the kind of death :)

**Comment by D3nnis3n** [comment]
Yeah, you could just allow an option that accessneeded is not active when killed by player or something like that. But i'm still wondering why anyone can open my grave after ~ 5 minutes despite the option set to -1 on my server oO But anyways, this is the best gravestone mod! :)

**Comment by Corail31** [comment]
i saw that you use a beta of sponge, isn't it better to use a stable version? and try it again?

**Comment by D3nnis3n** [comment]
I usually use the most recent version. I'll try it in a minute.

**Comment by D3nnis3n** [comment]
https://www.spongepowered.org/downloads/spongeforge/stable/1.10.2
This is the latest stable version of Sponge. There are no none-beta builds of Sponge.

**Comment by Corail31** [comment]
you should try my mod on forge server, but it's good to know it bugs on sponge.
[Forge server link](https://files.minecraftforge.net/maven/net/minecraftforge/forge/index_1.10.2.html)

**Comment by D3nnis3n** [comment]
SpongeForge is only a mod for Forge. It is an forge server with sponge as a installed mod. You seem to misunderstand how sponge works.

**Comment by Corail31** [comment]
yes i don't use a lot of mods on minecraft :)

**Comment by D3nnis3n** [comment]
Ah, okay. Well, when finished i usually have between 120 - 200 mods installed. Mostly technic mods and a handful of good additions like graves :)

**Comment by Corail31** [comment]
i got the solution, will be fixed and achievements'll work on server ;)

**Comment by Corail31** [comment]
@D3nnis3n This'll be validate soon on curse and fix your trouble with sponge :)
https://minecraft.curseforge.com/projects/corail-tombstone/files/2410518

**Comment by D3nnis3n** [comment]
This did indeed fix the problem with achievements, but not the key problem - that still persists unfortunatley :(

**Comment by Corail31** [comment]
check the config on client and on server ; could be the prob with key :)

**Comment by Corail31** [comment]
i'm going to try sponge too with the mod :)

**Comment by Corail31** [comment]
I think i'll be fixed with this and should avoid conflict on the key if you can try when you'll have time :)
[Link to beta 1.2.3](https://github.com/Corail31/trash/blob/master/tombstone-1.2.3-BETA-1.10.2.jar?raw=true)

**Comment by D3nnis3n** [comment]
I got no teleportation or death altering mods - i'm playing a technic pack. EnderIO is the only mod i got, that does anything like teleporting.

That Beta does indeed fix the problem. But i get 2 keys instead of one and both vanish after opening the tomb. Would be nice if that "aesthethic" problem could be fixed, but otherwise it is now working.

**Comment by Corail31** [comment]
Sorry D3nnis3n but i spent 10 hours yesterday on it.
The bug comes from another mod so i need to know which make trouble because there's no reason for my code to not work (as all tests work). 
The link i give you was a short test on priority of event but i need to know which is causing this. (or i won't patch because all is working, except for you)

**Comment by Corail31** [comment]
@D3nnis3n [Link Beta2](https://github.com/Corail31/trash/blob/master/tombstone-1.2.3-BETA-1.10.2.jar?raw=true)

**Comment by D3nnis3n** [comment]
I'd REALLY be glad to tell you the mod - but i can't. Since three hours i'm now switching mods around and testing them alone, and the problem is there with all of them and also none of them :(

Your second beta also gives me the Key, but also two of them - so nothing changed.
I see one in the respawn screen and get a second after pressing "respawn".

Please don't bother any longer, if you don't want to - two keys is better than none - and both go away when the tomb is opened.

**Comment by Corail31** [comment]
Yes, it's good to see that you have a nearly good result :) We'll find the bug, i just make tests :)
I know why you got 2 keys i'm adjusting it :)

**Comment by D3nnis3n** [comment]
Found a new bug:
After another death i didn't get a key - at least that is what i thought. But i had some in my quickbar, but they are invisible, i can only see them when pressing the thought-empty slot. That is freaky! :O

**Comment by Corail31** [comment]
yes i know why :) 
try this :
[Link beta3](https://github.com/Corail31/trash/blob/master/tombstone-1.2.3-BETA3-1.10.2.jar?raw=true)

**Comment by Corail31** [comment]
[Link beta4](https://github.com/Corail31/trash/blob/master/tombstone-1.2.4-BETA3-1.10.2.jar?raw=true)

**Comment by D3nnis3n** [comment]
Unfortunatley no change. :(

**Comment by Corail31** [comment]
on the last too? you got 2 keys oO

**Comment by D3nnis3n** [comment]
On the last one, yes. I didnt try beta3 - should i try that, too?

**Comment by Corail31** [comment]
no i'll try other thing :)

**Comment by D3nnis3n** [comment]
The first key is there in respawn screen, the second gets added when pressing respawn. Didn't notice any invisible keys anymore, though.

**Comment by Corail31** [comment]
[Link beta5](https://github.com/Corail31/trash/blob/master/tombstone-1.2.3-BETA5-1.10.2.jar?raw=true)

**Comment by D3nnis3n** [comment]
Still two of them. Sorry, i really don't want to bug you any further :(

**Comment by Corail31** [comment]
If some people can confirm this duplicate key bugs, i could try to investigate more on it.
I'll rename this topic as double grave key unconfirmed, if you're ok :)

**Comment by D3nnis3n** [comment]
Yep, thats fine.

**Comment by Corail31** [comment]
If you have ideas on the mod, don't hesitate to private me on curse website :)
You should the decorative grave and the scroll of recall that allow to teleport to decorative graves when haunted by soul :) perhaps you'll find bug in it too related to your server :)

**Comment by Corail31** [comment]
@D3nnis3n Btw, i saw above that you use the config 1.2.1 (config files are stored in their own folder in 1.2.2 ;) so if you want to open the grave directly you need to set decayTime to -1 on this config file, default is 1, so 5minutes^^)

**Comment by steven-archer** [comment]
Hello, sorry to 'dig' this up (Ha) but I have been experiencing this same issue as well.

I'm using Tombstone 1.2.2. And Forge 1.10.2-12.18.3.2281. No sponge. But I do have a lot of other mods installed. 

Its behaving just like D3nnis3n described. The key appears on the "Game over" screen, but then disappears after pressing "Respawn" and I cannot open the grave unless I switch to Creative.

**Comment by Corail31** [comment]
Hi :) Do you use baubles?

**Comment by D3nnis3n** [comment]
Hey, i'm so glad someone has the same problem!

I used Baubles too, but have uninstalled it some days ago, as i do not really need it. Should i retry something?

**Comment by Corail31** [comment]
* For recycler : there's a way to add x-million recipes in one time ;)
/recycler exportCraftingRecipes (the file is created in the config folder and contains all the reverse crafting recipes that are not always in recycler, it's explained on the recycler page.)
* For tombstone : the config folder is on server if you play on server (and in this folder, there's a folder tombstone where the files are)

**Comment by D3nnis3n** [comment]
I opened a bug report over at recycler.

For tombstone: I know. And in the serverconfig it's -1 by default.

**Comment by Corail31** [comment]
@D3nnis3n @steven-archer Could you try this version if it corrects your bug ?
[Lien](https://github.com/Corail31/trash/blob/master/tombstone-1.2.3-1.10.2.jar?raw=true)

**Comment by steven-archer** [comment]
My decayTime was defaulted to -1, is that normal?
Also I do not have Baubles installed.
I'll go ahead and try that version and your suggestions, and let you know what happens.

**Comment by Corail31** [comment]
-1 is for no decay, 0 is instant decay, 1 is 5min (2 is 10)

**Comment by Corail31** [comment]
baubles is working i try with it ;)

**Comment by steven-archer** [comment]
So while dead, I have only one key.
![2017-04-28_11 56 48](https://cloud.githubusercontent.com/assets/23382181/25536814/104c27e4-2c0a-11e7-833f-0d9879467902.png)
For a split second after respawning, I have no key. The world in front of me is still reloading.
![2017-04-28_11 56 50](https://cloud.githubusercontent.com/assets/23382181/25536836/28a8d9c2-2c0a-11e7-8135-2c215f6b1803.png)
Then I have two keys when the world finishes reloading (which for me takes only a fraction of a second, I had to spam the screenshot key just to get that middle one)
![2017-04-28_11 56 52](https://cloud.githubusercontent.com/assets/23382181/25536894/662c3de8-2c0a-11e7-9899-d9218a6d7f59.png)
They both go away when I unlock my grave, so that's fine. It's just a bit unusual. I can live with this though. This is progress!

**Comment by Corail31** [comment]
yes i'll change something and give you another link :)

**Comment by D3nnis3n** [comment]
That's the same i have. And also -1 works as 1 in my config file. Dunno why. Seems the "-" is just ignored.

**Comment by Corail31** [comment]
Can you try the link again? @D3nnis3n @steven-archer 
It's the same name but a different file.

**Comment by steven-archer** [comment]
Yep, that one gave the key before respawning, and it was gone after respawning.

**Comment by D3nnis3n** [comment]
You could post a complete mod list, so we can see which we got in common?

**Comment by steven-archer** [comment]
I spent a chunk of the afternoon removing mods one by one until I finally found the mod that seems to be interfering with the keys. A mod called [Levels](https://minecraft.curseforge.com/projects/levels?gameCategorySlug=mc-mods&projectID=235552) - As soon as I removed that, it started working normally again. The key stayed in my inventory. @D3nnis3n Do you use that mod too?

**Comment by Corail31** [comment]
only one key steven?

**Comment by D3nnis3n** [comment]
Haha, yes! But it is not in my main mod folder, i moved to an "test further" folder and therefore did not delete it when testing with mods - LOL. So it's the levels mod that makes the problems. You got it nailed down!

**Comment by D3nnis3n** [comment]
Do you mean this mod: https://minecraft.curseforge.com/projects/levels
I removed that some time ago, but i still got this mod: https://minecraft.curseforge.com/projects/level-up?gameCategorySlug=mc-mods&projectID=251136

**Comment by Corail31** [comment]
ok i had it to incompatibility mod in main page, thanks you very much for your tests !!!

**Comment by D3nnis3n** [comment]
Okay, i start doing a new test, i removed that mod just some days ago - Corail should work fine now, then.

**Comment by D3nnis3n** [comment]
Maybe we could poke that mod author?

**Comment by Corail31** [comment]
i'll see what he does on events on his code but is it levels or level up?

**Comment by Corail31** [comment]
i saw level up have reports about events priority

**Comment by D3nnis3n** [comment]
I do no longer use levels, only level up. I will test it right now with the default 1.2.2 version. I used both in my last tests.

**Comment by steven-archer** [comment]
On a totally different note, I don't know if this is a bug or not, but I noticed if I kill myself while I'm in creative, no tombstone will be made, and my entire inventory drops on the ground. Not sure if maybe this is just a minecraft thing.

**Comment by Corail31** [comment]
this is normal, creative is not a survival mode, you are not intended to die in creative :p

**Comment by D3nnis3n** [comment]
I can confirm the problematic mod is levels, it works fine now that i no longer use it. Level Up works fine. Great!

**Comment by Corail31** [comment]
Thanks for your help @D3nnis3n 
I won't fix compatibility for this mod so, i suppose i can close this issue.

**Comment by steven-archer** [comment]
Sounds fine. Glad I could help figure it out :)

**Comment by D3nnis3n** [comment]
@steven-archer
I decided to use "Quality Tools" instead, even before i knew that this mod created this bug.
